configuration/impl/src/main/java/org/jboss/forge/addon/configuration/proxy/ForgeProxySelector.java

```diff
@@ -81,12 +84,20 @@ private boolean isProxyAvailable(final URI uri)
       boolean isValidProtocol =
                "http".equalsIgnoreCase(protocol) || "https".equalsIgnoreCase(protocol);
 
-      boolean isHostExcluded =
-               proxySettings.getNonProxyHosts() != null
-                        && proxySettings.getNonProxyHosts().contains(host);
+      boolean isHostExcluded = proxySettings.getNonProxyHosts() != null && matches(host);
       return isValidProtocol && !isHostExcluded;
    }
 
+   private boolean matches(String target)
+   {
+      return proxySettings.getNonProxyHosts()
+         .stream()
+         .anyMatch(nonHost -> {
+            final PathMatcher pathMatcher = FileSystems.getDefault().getPathMatcher("glob:" + nonHost);
+            return pathMatcher.matches(Paths.get(target));
```

Comment:
Are you using a file matcher to match host names? Why not use `java.util.regex.Pattern` instead?

Reply:
This is the JDK implementation of Globs to Regexp: https://github.com/frohoff/jdk8u-jdk/blob/master/src/share/classes/sun/nio/fs/Globs.java

Reply:
Also instead of Glob perhaps it's better (and easier) to consider to be a Regexp instead

Reply:
Or if you still prefer to use the Path API (to avoid copying classes over), you can refactor the method as follow: 


```java
   boolean matchesHost(String target)
   {
      final boolean result;
      if (proxySettings.getNonProxyHosts() != null) {
         Path targetPath = Paths.get(target);
         FileSystem fileSystem = FileSystems.getDefault();
         result = proxySettings.getNonProxyHosts().stream().anyMatch(nonHost ->  {
            PathMatcher pathMatcher = (PathMatcher) fileSystem.getPathMatcher("glob:" + nonHost);
            return pathMatcher.matches(targetPath);
         });
      } else {
         result = false;
      }
      return result;
   }
```

Reply:
Although I am a bit worried about this strategy since you're just comparing Strings and `Path` tend to be resolved from the default `FileSystem` so I'd prefer not using it but using a `java.util.regex.Pattern` instead (which could be converted from a Glob pattern)

Reply:
I believe I decided because the dot (.) is **not** a special character in glob

Reply:
Glob to Regex with copied Globs.java file